activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1065,6 +1074,9 @@ def reverse_sql_order(order_query)
           when Arel::Nodes::Ordering
             o.reverse
           when String
+            # ensure we're not dealing with string subclass (Eg. Arel::Nodes::SqlLiteral)
+            o = String.new(o)
```

Comment:
Should we be undoing this afterwards?

Reply:
This is only used for setting `self.order_values`. I *think* it should be fine left as a `String`, but am not really sure. I feel like there might be more unintended consequences of instantiating a new copy of the original class with the resulting string.